builder/azure/arm/builder.go

```diff
@@ -83,13 +85,20 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 
 	b.setTemplateParameters(b.stateBag)
 
+	machineTemplateFactory := func(config *Config) (*resources.Deployment, error) {
+		return GetVirtualMachineDeployment(config)
+	}
+	keyVaultDeploymentFactory := func(config *Config) (*resources.Deployment, error) {
+		return GetKeyVaultDeployment(config)
+	}
+
 	var steps []multistep.Step
 
-	if b.config.OSType == constants.Target_Linux {
+	if strings.EqualFold(b.config.OSType, constants.Target_Linux) {
```

Comment:
I just learned about it too.  I came across while I was researching something else.

I also just learned about [x/net/context](https://godoc.org/golang.org/x/net/context) and [x/net/trace](https://godoc.org/golang.org/x/net/trace).  Great stuff !  I have some more refactoring to do based on this, especially around my interruptible task work.